### ISSUE TITLE
No handling for API throttling?

### ISSUE THREAD

**Issue by mfonsen** [takomo-io/takomo#78]
On a deploy I stumbled upon Cloudformation API throttling. More specifically with `DescribeStacks` operation.

Takomo fails after trying seven times by default. I can't find any special handling for to error code `Throttling` in the code base. Sceptre logic has been documented in [1]. It has proven to be very reliable but I'm not sure if this is the best approach.

### Version
**1.3.1**. I tried to reproduce the problem with 1.4.0 maybe due to some AWS automatic API limit tuning I can't reproduce the problem. 

https://github.com/takomo-io/takomo/blob/6104896c1b90654ddb0e63de2703a7327d997c85/packages/aws-clients/src/aws-client.ts#L39

[1] https://github.com/Sceptre/sceptre/blob/master/sceptre/connection_manager.py#L24

**Comment by mfonsen** [comment]
Happens also with 1.4.0

**Comment by hmeltaus** [comment]
I'll investigate and fix this. Thanks for the link to Sceptre implementation.

**Comment by hmeltaus** [comment]
AWS SDK has built-in throttling and that `maxRetries` property is used to control it. Increasing just the `maxRetries` would fix the throttling issue but the wait times would become too high.

I provided a custom backoff function for SDK that controls retrying and also increased max retries to 30. I was able to run larger deployments with this change without errors. Hopefully this will fix the throttling errors you are facing.

Current Takomo implementation makes a lot of API calls in parallel and when there are a higher number of stacks, this will cause exceeding of API rate limits. I need to make some adjustments on how API calls are made, maybe add some jitter and additional logic like queues to prevent too high request rates. I'll create a separate issue for this.

**Comment by hmeltaus** [comment]
Released version `1.4.1` with this fix. Will release fix to 2.x later.

**Comment by mfonsen** [comment]
Thank you! Upgraded to 1.4.1. Based on the logs I can confirm that this update makes deployments slower but avoids failing to throttle errors. Still I feel that Takomo is blazing fast compared to Sceptre.

`2020-08-20 14:03:46 +0300 [debug] - /killerapp/dev/eu-west-1/foo/bar.yml/eu-west-1 - Request limit exceeded, retry count: 8, pausing: 10817ms`

Personally I prefer waiting over aborting and retrying. It takes usually more time and might still fail due to throttling.

Improving the API call logic sounds like a great idea in order to avoid throttling in the first place.

**Comment by hmeltaus** [comment]
Created #79 for further work on this topic
